test/e2e/testdata/artifact-repository-ref.yaml

```diff
@@ -3,25 +3,43 @@ kind: Workflow
 metadata:
   name: artifact-repository-ref
   labels:
-    argo-e2e: true
+    argo-e2e: "true"
 spec:
   entrypoint: main
   artifactRepositoryRef:
     key: minio
   templates:
     - name: main
+      dag:
+        tasks:
+          - name: generate
+            template: generate
+          - name: consume
+            template: consume
+            dependencies:
+              - generate
+    - name: generate
       container:
         image: argoproj/argosay:v2
         args:
           - echo
           - hello
-          - /tmp/file
-      inputs:
+          - /mnt/file
+      outputs:
         artifacts:
           - name: file
-            path: /tmp/file
-            optional: true
-      outputs:
+            path: /mnt/file
+            s3:
```

Comment:
I wonder whether it's possible to simplify this further to a common `uri` field:
```
    uri: s3://my-bucket/my-file
```

Reply:
something like go-getter?